### PR TITLE
implement transformPartitionValues for custom transformations

### PR DESCRIPTION
**Pull request by zzeekk**
see #264

### CODE REVIEW

File: sdl-core/src/main/scala/io/smartdatalake/workflow/action/customlogic/CustomDfTransformer.scala [hunk 1]
```diff
@@ -44,6 +44,16 @@ trait CustomDfTransformer extends Serializable {
    */
   def transform(session: SparkSession, options: Map[String, String], df: DataFrame, dataObjectId: String) : DataFrame
 
+  /**
+   * Optional function to define the transformation of input to output partition values.
+   * For example this enables to implement aggregations where multiple input partitions are combined into one output partition.
+   * Note that the default value is input = output partition values, which correct for most use cases.
```
Comment (pgruetter) [inline sdl-core/src/main/scala/io/smartdatalake/workflow/action/customlogic/CustomDfTransformer.scala:50]:
```suggestion
   * Note that the default value is input = output partition values, which should be correct for most use cases.
```

File: sdl-core/src/main/scala/io/smartdatalake/workflow/action/customlogic/CustomDfsTransformer.scala [hunk 2]
```diff
@@ -42,6 +42,15 @@ trait CustomDfsTransformer extends Serializable {
    */
   def transform(session: SparkSession, options: Map[String,String], dfs: Map[String,DataFrame]) : Map[String,DataFrame]
 
+  /**
+   * Optional function to define the transformation of input to output partition values.
+   * For example this enables to implement aggregations where multiple input partitions are combined into one output partition.
+   * Note that the default value is input = output partition values, which correct for most use cases.
```
Comment (pgruetter) [inline sdl-core/src/main/scala/io/smartdatalake/workflow/action/customlogic/CustomDfsTransformer.scala:48]:
```suggestion
   * Note that the default value is input = output partition values, which should be correct for most use cases.
```

File: sdl-core/src/test/scala/io/smartdatalake/workflow/action/CustomSparkActionTest.scala [hunk 3]
```diff
@@ -285,7 +285,7 @@ class CustomSparkActionTest extends FunSuite with BeforeAndAfter {
     val srcSubFeed = SparkSubFeed(None, "src1", Seq())
     val tgtSubFeed = action1.exec(Seq(srcSubFeed)).head
 
-    session.table(s"${tgtTable1.fullName}").show
+    //session.table(s"${tgtTable1.fullName}").show
```
Comment (pgruetter) [inline sdl-core/src/test/scala/io/smartdatalake/workflow/action/CustomSparkActionTest.scala:288]: Please remove or uncomment

**Review by pgruetter — APPROVED** [review]
Just minor comments

**Comment by zzeekk** [comment]
Sorry, missed to fix your comments in this branch. I'll fix them in feature/268-custom-execution-mode

**Pull request merged**